src/runtime/propensity-server-test.js

```diff
@@ -296,4 +312,132 @@ describes.realWin('PropensityServer', {}, env => {
+    expect(receivedContext).to.equal(str);
+
+    //now ensure it properly converts each analytics event it receives into
+    //an appropriate propensity event (or suppresses the event entirely if it)
+    //isn't a propensity event.
+    for (const k in AnalyticsEvent) {
+      defaultEvent.eventType = AnalyticsEvent[k];
+      receivedType = null;
+      propensityServer.sendClientEvent_(defaultEvent);
+      switch (defaultEvent.eventType) {
+        case AnalyticsEvent.UNKNOWN:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.IMPRESSION_PAYWALL:
+          expect(receivedType).to.equal(PropensityApi.Event.IMPRESSION_PAYWALL);
+          break;
+        case AnalyticsEvent.IMPRESSION_AD:
+          expect(receivedType).to.equal(PropensityApi.Event.IMPRESSION_AD);
+          break;
+        case AnalyticsEvent.IMPRESSION_OFFERS:
+          expect(receivedType).to.equal(PropensityApi.Event.IMPRESSION_OFFERS);
+          break;
+        case AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.IMPRESSION_SMARTBOX:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_SUBSCRIBE:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_PAYMENT_COMPLETE:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_PAYMENT_COMPLETED);
+          break;
+        case AnalyticsEvent.ACTION_ACCOUNT_CREATED:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_ACCOUNT_ACKNOWLEDGED:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_SUBSCRIPTIONS_LANDING_PAGE:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_SUBSCRIPTIONS_LANDING_PAGE);
+          break;
+        case AnalyticsEvent.ACTION_PAYMENT_FLOW_STARTED:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_PAYMENT_FLOW_STARTED);
+          break;
+        case AnalyticsEvent.ACTION_OFFER_SELECTED:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_OFFER_SELECTED);
+          break;
+        case AnalyticsEvent.EVENT_PAYMENT_FAILED:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.EVENT_CUSTOM:
+          expect(receivedType).to.equal(PropensityApi.Event.EVENT_CUSTOM);
+          break;
+        default:
+          expect(false).to.be.true; //add new analytics event to the switch
+      }
+    }
+  });
+
+  it('should respect the SwG logging experiment flag', () => {
+    let receivedType = null;
+    let receivedContext = null;
+    sandbox.stub(PropensityServer.prototype, 'sendEvent',
+        (eventType, context) => {
+          receivedType = eventType;
+          receivedContext = context;
+        });
+
```

Comment:
Better to avoid testing the internal implementation if there's another way.

Reply:
I think this is done

Reply:
you are still calling handleClientEvent_() which is a internal private method.

You need to capture the callback registered if you want to invoke the listener.

Reply:
Now I think I know what you mean.  Done?

Reply:
just realized one more thing. since you are programming is_active field into the event parameters based on active field of the PropensityEvent, it's better to test that as well.

Reply:
Done